cli/azd/pkg/project/framework_service_maven.go

```diff
@@ -153,6 +174,58 @@ func (m *mavenProject) Package(
 	}, nil
 }
 
+// funcAppDir returns the directory of the function app packaged by azure-functions-maven-plugin for the given service.
+//
+// The app is typically packaged under target/azure-functions.
+func (m *mavenProject) funcAppDir(ctx context.Context, svc *ServiceConfig) (string, error) {
+	svcPath := svc.Path()
+	// The staging directory for azure-functions-maven-plugin is target/azure-functions.
+	// It isn't configurable, but this may change in the future: https://github.com/microsoft/azure-maven-plugins/issues/1968
+	functionsStagingRel := filepath.Join("target", "azure-functions")
+	functionsStagingDir := filepath.Join(svcPath, functionsStagingRel)
+
+	// A conventional azure-functions-maven-plugin project will have the property 'functionAppName' in pom.xml,
+	// with its property value is passed to azure-functions-maven-plugin as 'appName'.
+	appName, err := m.mavenCli.GetProperty(ctx, "functionAppName", svcPath)
+	if err != nil && !errors.Is(err, maven.ErrPropertyNotFound) {
+		return "", err
+	}
+
+	if appName != "" {
+		funcDir := filepath.Join(functionsStagingDir, appName)
+		if _, err := os.Stat(funcDir); err == nil {
+			return funcDir, nil
+		} else if !errors.Is(err, os.ErrNotExist) {
+			return "", fmt.Errorf("checking for function app directory: %w", err)
+		}
+	}
+
+	entries, err := os.ReadDir(functionsStagingDir)
+	if err != nil {
+		return "", fmt.Errorf("reading azure-functions directory: %w", err)
+	}
+
+	dirs := make([]string, 0, len(entries))
+	for _, entry := range entries {
+		if entry.IsDir() {
+			dirs = append(dirs, entry.Name())
+		}
+	}
+
+	if len(dirs) == 1 {
+		return filepath.Join(functionsStagingDir, dirs[0]), nil
+	} else {
+		for i := range dirs {
+			dirs[i] = filepath.Join(functionsStagingRel, dirs[i])
+		}
+
+		return "", fmt.Errorf(
+			//nolint:lll
+			"multiple staging directories found: %s. Specify 'dist' in azure.yaml to select a specific directory",
+			strings.Join(dirs, ", "))
```

Comment:
nit. consider to use `ListAsText()` here

Reply:
I did intentionally not use `ListAsText` when writing this, and still have the same leanings after trying both out.

The user isn't going to quite understand or be familiar with these paths as they're largely implementation details. To that end, treating it as comma-delimited data to parse out, rather than an English sentence makes more sense to me.